services/director-v2/tests/unit/with_dbs/test_modules_redis.py

```diff
@@ -0,0 +1,279 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import asyncio
+from argparse import Namespace
+from asyncio import CancelledError, Task
+from contextlib import suppress
+from typing import AsyncIterable, Optional
+
+import pytest
+from asgi_lifespan import LifespanManager
+from faker import Faker
+from fastapi import FastAPI
+from redis.asyncio import Redis
+from redis.asyncio.lock import Lock
+from redis.exceptions import LockError
+from settings_library.redis import RedisSettings
+from simcore_service_director_v2.modules import redis
+from simcore_service_director_v2.modules.redis import (
+    EXTEND_TASK_ATTR_NAME,
+    DockerNodeId,
+    RedisLockManager,
+    auto_release,
+)
+
+pytest_simcore_core_services_selection = [
+    "redis",
+]
+
+# UTILS
+
+
+class MockLocksPerNodeProvider:
+    def __init__(self, lock_per_node: int) -> None:
+        self.lock_per_node = lock_per_node
+
+    async def get(self, *args, **kwargs):
+        return self.lock_per_node
+
+
+# FIXTURES
+
+
+@pytest.fixture
+async def minimal_app(redis_settings: RedisSettings) -> AsyncIterable[FastAPI]:
+    app = FastAPI()
+
+    # add expected redis_settings
+    app.state.settings = Namespace()
```

Comment:
Why not just overriding environment variable fixture to change the configuration and use normal app fixture?
Why yet another minimal_app.